src/components/slideshow/slideshow.js

```diff
@@ -148,112 +179,134 @@ define(['dialogHelper', 'inputManager', 'connectionManager', 'layoutManager', 'f
                 if (btnDownload) {
                     btnDownload.addEventListener('click', download);
                 }
 
-                var btnShare = dlg.querySelector('.btnShare');
+                var btnShare = dialog.querySelector('.btnShare');
                 if (btnShare) {
                     btnShare.addEventListener('click', share);
                 }
             }
 
             setUserScalable(true);
 
-            dialogHelper.open(dlg).then(function () {
-
+            dialogHelper.open(dialog).then(function () {
                 setUserScalable(false);
-                stopInterval();
             });
 
             inputManager.on(window, onInputCommand);
             document.addEventListener((window.PointerEvent ? 'pointermove' : 'mousemove'), onPointerMove);
 
-            dlg.addEventListener('close', onDialogClosed);
+            dialog.addEventListener('close', onDialogClosed);
 
             if (options.interactive) {
-                loadSwiper(dlg);
+                loadSwiper(dialog, options);
             }
         }
 
+        /**
+         * Handles OSD changes when the autoplay is started.
+         */
         function onAutoplayStart() {
-            var btnSlideshowPause = dlg.querySelector('.btnSlideshowPause i');
+            var btnSlideshowPause = dialog.querySelector('.btnSlideshowPause i');
             if (btnSlideshowPause) {
-                btnSlideshowPause.classList.remove("play_arrow");
-                btnSlideshowPause.classList.add("pause");
+                btnSlideshowPause.classList.replace("play_arrow", "pause");
             }
         }
 
+        /**
+         * Handles OSD changes when the autoplay is stopped.
+         */
         function onAutoplayStop() {
-            var btnSlideshowPause = dlg.querySelector('.btnSlideshowPause i');
+            var btnSlideshowPause = dialog.querySelector('.btnSlideshowPause i');
             if (btnSlideshowPause) {
-                btnSlideshowPause.classList.remove("pause");
-                btnSlideshowPause.classList.add("play_arrow");
+                btnSlideshowPause.classList.replace("pause", "play_arrow");
             }
         }
 
-        function loadSwiper(dlg) {
-
+        /**
+         * Initializes the Swiper instance and binds the relevant events.
+         * @param {HTMLElement} dialog - Element containing the dialog.
+         * @param {Object} options - Options used to initialize the Swiper instance.
+         */
+        function loadSwiper(dialog, options) {
+            var slides;
             if (currentOptions.slides) {
-                dlg.querySelector('.swiper-wrapper').innerHTML = currentOptions.slides.map(getSwiperSlideHtmlFromSlide).join('');
+                slides = currentOptions.slides;
             } else {
-                dlg.querySelector('.swiper-wrapper').innerHTML = currentOptions.items.map(getSwiperSlideHtmlFromItem).join('');
+                slides = currentOptions.items;
             }
```

Comment:
cannot you do just `var slides = currentOptions.slides || currentOptions.items`?

Reply:
Yeah, it should work.